allow install via composer

daemon-byte opened the pull request:
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tested the changes


## Pull request type

<!-- Please do not submit updates to dependencies unless it fixes an issue. --> 

<!-- Please try to limit your pull request to one type, submit multiple pull requests if needed. --> 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Other (please describe): 


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
you have to download a tarball and unzip it
Issue Number: N/A


## What is the new behavior?
<!-- Please describe the behavior or changes that are being added by this PR. -->
you now use composer require
-
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [x ] No

<!-- If this introduces a breaking change, please describe the impact and migration path for existing applications below. -->
It's possible the old method doesn't work anymore. I can't test that with my setup but it should be easily fixed if that's the case.
## Docs need to be updated?

- [ ] Yes
- [x] No

<!-- If this introduces a doc update, please describe it below. -->
I already updated it
## Other information

<!-- Any other information that is important to this PR such as screenshots of how the component looks before and after the change. -->

padams commented:
@daemon-byte It looks like a commit slipped in before this. can you resolve the conflict? I'll merge today.

daemon-byte commented:
I pulled your master into mine and dealt with the conflict so that should work now? I'm new to contributing back to forks

padams commented:
@daemon-byte  No worries. you are doing great. one more change is needed though. 

'''owa-autoloader.php''' is needed for people that do not use composer to install this SDK. So we actually need to leave that file where it is and not move it into the '''src/''' directory. Also the class paths in that file need to be changed to reflect the new location of the files or... i suppose we could remove that autoloader entirely now that composer is doing the full autoloading.

daemon-byte commented:
It's not needed because of composer but let me have a quick play with it today. Maybe I can find a way to keep the legacy method as an option without pushing people into using composer. Options are good

padams commented:
Yeah i think its important for backward compatibility.

daemon-byte commented:
ok so I moved the file back and updated it so you can use the old method or ignore the file and use composer. My test syytem was php 8.0 so I fixed a couple of depreciated errors as well. There was also 1 function where the group and name were the wrong way around vs the documentation so I fixed that as well.

padams commented on src/Tracker/TrackerClient.php:
```diff
@@ -447,7 +447,7 @@ public function trackPageview($event = '') {
     }
 
-    public function trackAction($action_group = '', $action_name, $action_label = '', $numeric_value = 0) {
+    public function trackAction($action_name, $action_group = '', $action_label = '', $numeric_value = 0) {
```
We should fix the docs to match the code, not the other way around as this would be a breaking change for people.

daemon-byte replied:
Yeh I ummed over it. Do you fix the bug and break code using the bug or leave it as is. I was going with leave it but in 8.0 this will break. So at some point you need to change it

padams replied:
Oh I missed that. Are you saying that in php8 the setting of the default as the first parameter throws an error?

daemon-byte replied:
In 8.0 you get a depreciated warning if you have a required parameter after an optional one.

padams replied:
I see. Let's leave it the way you fixed it then and we will note the change. Both are strings so it won't throw an error but will change the labels on action events.

padams commented on src/Tracker/State.php:
```diff
@@ -105,7 +105,7 @@ public function get($store, $name = '') {
         }
     }
 
-    function setState($store, $name = '', $value, $store_type = '', $is_perminent = false) {
```
The default is needed as one could set a store without a name.

daemon-byte replied:
I didn't get any errors like that. I couldn't find anywhere those variables weren't set already.
